src/net_processing.cpp

```diff
@@ -1833,8 +1833,7 @@ bool PeerManagerImpl::MaybePunishNodeForBlock(NodeId nodeid, const BlockValidati
         return true;
     // Conflicting (but not necessarily invalid) data or different policy:
     case BlockValidationResult::BLOCK_MISSING_PREV:
-        // TODO: Handle this much more gracefully (10 DoS points is super arbitrary)
-        if (peer) Misbehaving(*peer, 10, message);
+        if (peer) Misbehaving(*peer, 100, message);
```

Comment:
This is only triggered by `AcceptBlockHeader`, which typically happens _after_ `if(!headers_connect_blockindex)` causes an early return in `ProcessHeadersMessage`.

In other words, disconnecting here does not get in the way of `HandleUnconnectingHeaders` processing.

Presumably (I didn't check) this can be reached if we get an unsolicited BLOCK message. In which case it seems fine to disconnect.

It can also be reached when using the `submitblock(header)` RPC, but in that case there is no `peer`.